app/routes/surveys/EditSurveyRoute.js

```diff
@@ -0,0 +1,47 @@
+import { connect } from 'react-redux';
+import prepare from 'app/utils/prepare';
+import { compose } from 'redux';
+import { editSurvey, fetch, deleteSurvey } from '../../actions/SurveyActions';
+import SurveyEditor from './components/SurveyEditor';
+import { LoginPage } from 'app/components/LoginForm';
+import replaceUnlessLoggedIn from 'app/utils/replaceUnlessLoggedIn';
+import { selectSurveyById } from 'app/reducers/surveys';
+import { push } from 'react-router-redux';
+import loadingIndicator from 'app/utils/loadingIndicator';
+
+const mapStateToProps = (state, props) => {
+  const surveyId = Number(props.params.surveyId);
+  const survey = selectSurveyById(state, { surveyId });
+  const fetching = state.surveys.fetching;
```

Comment:
might as well just `fetching: state.surveys.fetching` in the object (or at least use destructuring)